File: comgongdaeng/TwoPointer/BOJ_2003.py
import sys

N, M = map(int, sys.stdin.readline().split())
sequence = list(map(int, sys.stdin.readline().split()))

sum = sequence[0]
left = cnt = 0
right = 1
while left <= right and right <= N:
    if sum == M:
        cnt += 1
        sum -= sequence[left]
        left += 1
    elif sum < M:
            if right < N:
                sum += sequence[right]
                right += 1
            else:
                 break
    else:
        sum -= sequence[left]
        left += 1

print(cnt)

File: keemsebin/TwoPointer/BOJ_2003.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let [N, M] = input[0].split(" ").map(Number);
  let arr = input[1].split(" ").map(Number);
  let ans = 0;
  for (let i = 0; i < N; i++) {
    let sum = 0;
    for (let j = i; j < N; j++) {
      sum += arr[j];
      if (sum === M) {
        ans++;
        break;
      } else if (sum > M) {
        break;
      }
    }
  }
  console.log(ans);
});
